Task/grow 390 messaging for channel input

daisymarie128 opened the pull request:
This PR adds new ui messaging to the channel counter.

Jira ticket: https://buffer.atlassian.net/browse/GROW-390

Currently, still a draft as I haven't been able to test all the scenarios which could break the submit button with the new code added.


<img width="1309" alt="Screen Shot 2022-03-03 at 3 17 00 pm" src="https://user-images.githubusercontent.com/7763710/156504048-ab733c55-056a-4d60-839d-8f2c9f91fa8a.png">
<img width="1311" alt="Screen Shot 2022-03-03 at 3 16 40 pm" src="https://user-images.githubusercontent.com/7763710/156504053-7fb5b929-7ab2-4ea9-881e-bbec99c308ff.png">

<img width="1310" alt="Screen Shot 2022-03-03 at 3 17 09 pm" src="https://user-images.githubusercontent.com/7763710/156504045-45e949e0-3b3e-4ffc-a845-330f9f8d044b.png">



<img width="1313" alt="Screen Shot 2022-03-03 at 3 17 17 pm" src="https://user-images.githubusercontent.com/7763710/156504033-e9641b02-72f0-4f3e-a26b-9d7ee72673f2.png">

dinostheo commented on packages/app-shell/src/components/Modal/modals/PlanSelector/hooks/useChannelsCounter.js:
```diff
@@ -10,20 +25,54 @@ const useChannelsCounter = (initialChannelCount = 0, minimumQuantity = 1) => {
+
+    if (
+      planId === 'agency' &&
+      channelsCount === 10 &&
```
This will work but we should use the `minimunQuantity` of the `channelSlotDetails`, [see here](https://github.com/bufferapp/core-authentication-service/blob/main/core-api/src/graphql/schema.graphql#L348). In that way, we avoid hardcoding values and use a similar code, for every possible plan. Also, we have noticed in the past that we tend to experiment and change those values. It is less error-prone if we are having a single source of truth for those values. 

Regarding the free plan, we could also use the hard limit of the plan.

daisymarie128 replied:
Nice one @dinostheo , I've updated this now, but I just wasn't sure about the Free plan. What did you mean to use the hard limit of the plan? Where would I find this value? I couldn’t find it in the plan details so have left it hard coded for now.

dinostheo commented on packages/app-shell/src/components/Modal/modals/PlanSelector/hooks/useChannelsCounter.js:
```diff
@@ -10,20 +25,54 @@ const useChannelsCounter = (initialChannelCount = 0, minimumQuantity = 1) => {
+
+    if (
+      planId === 'free' &&
+      channelsCount <= 4 &&
```
Shouldn't this value be `< 4` or `<= 3`? 
Also I realize that we might want to add the `limits` on the plan options as well. https://github.com/bufferapp/core-authentication-service/blob/main/core-api/src/graphql/schema.graphql#L131

dinostheo replied:
@daisymarie128 this value is already available under the linked structure. However, I think it would be better for our use case if we would also add it to the channel details.

daisymarie128 commented on packages/app-shell/src/components/Modal/utils.js:
```diff
@@ -114,6 +114,21 @@ export function calculateTotalSlotsPrice(
+export function getAgencyPlanMinimumChannelInputMessaging() {
+  return {
+    messageStatus: 'warning',
+    message: 'Oops! The Agency Plan has a minimum of 10 channels. ',
```
Oh interesting 🤔 This is how it's setup to show now.

Have the Agency plan selected. 
When you have 10 channels selected and then try to click the - again it should show up.
It will then disappear if you increase the channel slots.

Though I'm not sure if this interaction is perfect because the message will still stay there if you leave it on 10 channels. 🤔

daisymarie128 commented:
<img width="1307" alt="Screen Shot 2022-03-07 at 11 54 31 am" src="https://user-images.githubusercontent.com/7763710/156954663-a08e6737-d452-4acb-801b-c3a0e1c295a4.png">
<img width="1309" alt="Screen Shot 2022-03-07 at 11 54 12 am" src="https://user-images.githubusercontent.com/7763710/156954683-7ea5132f-3d69-4e2a-b425-c41c077bc7ef.png">

Latest updates addressing the comments above.
Also added some styling fixes to ensure the summary text doesn't flow behind the submit button.
All the code should now be behind the FF so I will merge this now :)
